native/src/modules/endpoint/sagas/watchFetchEvent.js

```diff
@@ -26,8 +32,11 @@ export function * fetchEvent (dataContainer: DataContainer, action: FetchEventAc
         call(dataContainer.getResourceCache, city, language)
       ])
 
-      const insert: PushEventActionType = {
-        type: 'PUSH_EVENT',
+      const lastUpdate: Moment | null = yield call(dataContainer.getLastUpdate, city, language)
+
+      // $FlowFixMe Flow can't evaluate the type as it is dynamic
```

Comment:
That actually should work. I think it is possible to refactor this that it works. You can create a function with return type PushCategoryActionType | RefreshCategoryActionType. Inside that function you have a if which returns one action or the other one.